core/types/calls.cc

```diff
@@ -1338,9 +1358,20 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, const DispatchArgs &arg
                 }
                 NameRef arg = key.asName();
 
-                if (auto e = gs.beginError(args.callLoc(), errors::Infer::MethodArgumentCountMismatch)) {
+                auto kwargErrLoc = kwargsLoc;
+                auto it = kwargLocs.find(key.asName());
+                // TODO(jez) This papers over some stuff around kwsplats which might get in our way
+                // of getting a loc for known keyword args. In those cases, we simply give up right now.
+                if (it != kwargLocs.end()) {
+                    kwargErrLoc = it->second;
+                }
+                if (auto e = gs.beginError(kwargErrLoc, errors::Infer::MethodArgumentCountMismatch)) {
                     e.setHeader("Unrecognized keyword argument `{}` passed for method `{}`", arg.show(gs),
                                 method.show(gs));
+                    if (kwargErrLoc != kwargsLoc && kwargErrLoc.exists()) {
```

Comment:
Nit: this seems slightly clearer with the condition reversed (s/o to `std::optional` enabling comparison to non-`optional` types).

```suggestion
                    if (kwargErrLoc.exists() && kwargErrLoc != kwargsLoc) {
```

Reply:
I don't think there's an `optional` in play here, were you suggesting that I introduce it somehow?